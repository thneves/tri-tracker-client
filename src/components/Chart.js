import React from 'react';
import PropTypes from 'prop-types';
import { Bar } from 'react-chartjs-2';
import options from '../helpers/chartOptions';

const Chart = ({
  months, runnings, swimmings, cyclings,
}) => {
  const data = {
    labels: months,
    datasets: [
      {
        label: '# Runnings',
        data: runnings,
        backgroundColor: [
          'rgba(0, 186, 255, 1)',
        ],
        borderColor: [
          'rgba(0, 186, 255, 1)',

        ],
        borderWidth: 1,
      },
      {
        label: '# Cycling',
        data: cyclings,
        backgroundColor: [
          'rgba(19, 59, 74, 1)',
        ],
        borderColor: [
          'rgba(19, 59, 74, 1)',
        ],
        borderWidth: 1,
      },
      {
        label: '# Swimmings',
        data: swimmings,
        backgroundColor: [
          'rgba(9, 80, 188, 1)',
        ],
        borderColor: [
          'rgba(9, 80, 188, 1)',
        ],
        borderWidth: 1,
      },
    ],
  };

  return (
    <Bar
      data={data}
      options={options}
      height={80}
      width={160}
    />
  );
};

Chart.propTypes = {
  months: PropTypes.instanceOf(Array).isRequired,
  runnings: PropTypes.instanceOf(Array).isRequired,
  swimmings: PropTypes.instanceOf(Array).isRequired,
  cyclings: PropTypes.instanceOf(Array).isRequired,
};

export default Chart;
