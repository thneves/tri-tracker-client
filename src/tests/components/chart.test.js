import React from 'react';
import renderer from 'react-test-renderer';
import Chart from '../../components/Chart';

const months = ['Mar 2048', 'Dec 2034', 'Sep 2035'];
const runs = [1, 1, 1];
const swims = [1, 2, 4];
const cyclings = [2, 2, 2];

it('Should render the Chart component', () => {
  const chart = renderer
    .create(
      <Chart
        months={months}
        runnings={runs}
        swimmings={swims}
        cyclings={cyclings}
      />,
    ).toJSON();

  expect(chart).toMatchSnapshot();
});
