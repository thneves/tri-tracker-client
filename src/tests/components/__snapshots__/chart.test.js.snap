// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the Chart component 1`] = `
<canvas
  data-testid="canvas"
  height={150}
  onClick={[Function]}
  role="img"
  width={300}
/>
`;
